src/Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -383,5 +384,27 @@ public static bool IsImplementationOfAnyExplicitInterfaceMember(this ISymbol sym
 
             return false;
         }
+
+        /// <returns>
+        /// Returns true if symbol is a local variable and its declaring syntax node is 
+        /// after the current position, false otherwise (including for non-local symbols)
+        /// </returns>
+        public static bool IsInaccessibleLocal(this ISymbol symbol, int position)
```

Comment:
This will only work for structs, as they're initialized at declaration; class references will have definite assignment issues. Please update the method to something like `IsInaccessibleLocalStruct`, and in the method make sure that the passed symbol is of struct type.